test/functional/sentinel.js

```diff
@@ -186,7 +185,25 @@ describe('sentinel', function () {
         name: 'master'
       });
     });
+    it('should connect to sentinel with authentication successfully', function (done) {
+      var sentinel = new MockServer(27379, function (argv) {
+        if (argv[0] === 'auth' && argv[1] === 'pass') {
+          authed = true;
+        }
```

Comment:
Missing assertion here. Should define `authed` as `false` in the outer block, and add an assertion of `expect(authed).to.eql(true)` when `argv[0] === 'sentinel' && argv[1] === 'get-master-addr-by-name'` to make sure ioredis sends `AUTH` before asking for master address.

Close the connections here instead of on `sentinel.once('connect')`.

Reply:
Simply make sure the test will fail when there's a typo in option name 😄